### PR TITLE
Add a MR Unit Cell MOOSE MultiApps Example

### PR DESCRIPTION
**Pull request by miaoyinb**
The new example 6 include a SS+trN+tr approach for unit cell simulation.

### CODE REVIEW

**Comment by miaoyinb** [comment]
@nstauff FYI

File: examples/example6_adv_multiapps/example6.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+# SPDX-FileCopyrightText: 2022 UChicago Argonne, LLC
+# SPDX-License-Identifier: MIT
+
+from math import cos, pi
```
Comment (nstauff) [inline examples/example6_adv_multiapps/example6.py]: is "cos" and "pi" used in the script?

File: examples/example6_adv_multiapps/example6.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+for key in moose_result_ss.csv_data:
+    print(key, moose_result_ss.csv_data[key])
+print(moose_result_ss.inputs)
+print(moose_result_ss.outputs)
+
+# set up an enviroment variable to pass SS output path to trN and tr workflow
+os.environ["SS_PATH"] = str(moose_result_ss.base_path)
+
+# Null Transient Parameters
+params_trN = params_ss
```
Comment (nstauff) [inline examples/example6_adv_multiapps/example6.py:54]: not clear to me why we need to define a new `params_trN` and `params_tr`. If those databases contain the same information, let's just define it once as `params`

**Review by nstauff — CHANGES_REQUESTED** [review]
@miaoyinb thanks for developing this example. 
It looks good, but I am wondering if we could not further improve. My main motivation would be to remove completely overlap between the different inputs through templating. Currently, we still need 3 templated inputs, and 6 non-templated inputs - could we reduce to 3 inputs (Griffin/Bison/Sockeye) and template them in a way we could handle `ss`, `trN` and `tr` conditions?

File: examples/example6_adv_multiapps/example6.py (hunk 1 above)
Comment (miaoyinb) [reply examples/example6_adv_multiapps/example6.py]: Copied from template. will remove them.

File: examples/example6_adv_multiapps/example6.py (hunk 2 above)
Comment (miaoyinb) [reply examples/example6_adv_multiapps/example6.py:54]: I was trying to have some flexibility to allow different params for ss/trN/tr in the future.

**Comment by miaoyinb** [comment]
> @miaoyinb thanks for developing this example. It looks good, but I am wondering if we could not further improve. My main motivation would be to remove completely overlap between the different inputs through templating. Currently, we still need 3 templated inputs, and 6 non-templated inputs - could we reduce to 3 inputs (Griffin/Bison/Sockeye) and template them in a way we could handle `ss`, `trN` and `tr` conditions?

I agree with you. I am trying to do several incremental improvements to achieve that eventually. I am also thinking about leveraging MOOSE's "new" multiple input files support to do this (https://mooseframework.inl.gov/newsletter/2021_06.html).

File: examples/example6_adv_multiapps/example6.py [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+params_ss['BISON_Inifinit_Temperature'] = 800.0 # K.
+params_ss['BISON_Outside_HTC'] = 100.0
+
+# Sockeye
+params_ss['Sockeye_Evap_Elem_Num'] =15
+params_ss['Sockeye_Adia_Elem_Num'] =5
+params_ss['Sockeye_Cond_Elem_Num'] =10
+
+params_ss.show_summary(show_metadata=False, sort_by='key')
+
```
Comment (nstauff) [inline examples/example6_adv_multiapps/example6.py:39]: Let's add:
`print("Steady-state calculation")`

File: examples/example6_adv_multiapps/example6.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+print(moose_result_ss.inputs)
+print(moose_result_ss.outputs)
+
+# set up an enviroment variable to pass SS output path to trN and tr workflow
+os.environ["SS_PATH"] = str(moose_result_ss.base_path)
+
+# Null Transient Parameters
+params_trN = params_ss
+
+# MOOSE Workflow for Null transient
```
Comment (nstauff) [inline examples/example6_adv_multiapps/example6.py:56]: Let's add:
print("Null transient calculation")

File: examples/example6_adv_multiapps/example6.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+rel_diff_tol = 1.0e-8
+
+power_rel_diff = (moose_result_trN.csv_data['integrated_power'][0] - moose_result_trN.csv_data['integrated_power'][-1])/moose_result_trN.csv_data['integrated_power'][0]
+if power_rel_diff > rel_diff_tol:
+    raise RuntimeError("The Null Transient Run has a transient power; please check the steady state run.")
+
+# Transient Parameters
+params_tr = params_ss
+
+# MOOSE Workflow for transient
```
Comment (nstauff) [inline examples/example6_adv_multiapps/example6.py:76]: Let's add:
print("Transient calculation")

**Comment by miaoyinb** [comment]
I think it necessary to add colorful messages (of course in a separate PR in the future).

**Review by nstauff — APPROVED** [review]

**Pull request merged**